security/src/main/java/org/kaazing/gateway/security/auth/BasicLoginModule.java

```diff
@@ -61,16 +61,25 @@ public void initialize(Subject subject, CallbackHandler callbackHandler,
     protected boolean doLogin() throws LoginException {
 
         if (!authenticationSchemeIsBasic()) {
-            return true;
+            throw new LoginException("Only able to handle Basic authentication scheme");
```

Comment:
Yes, you are correct. The authentication process ignores the realms in the chain that already passed authentication. So, for a realm/chain of login modules, we can only have one challenge scheme. Returning false would also be correct.